rmw_connext_cpp/package.xml

```diff
@@ -14,22 +14,22 @@
   <buildtool_export_depend>rosidl_cmake</buildtool_export_depend>
 
   <build_depend>connext_cmake_module</build_depend>
-  <build_depend>libndds52-dev</build_depend>
   <build_depend>rcutils</build_depend>
   <build_depend>rmw</build_depend>
   <build_depend>rmw_connext_shared_cpp</build_depend>
   <build_depend>rosidl_generator_c</build_depend>
   <build_depend>rosidl_generator_cpp</build_depend>
+  <build_depend>rosidl_generator_dds_idl</build_depend>
```

Comment:
This move (from L25) comes from alpha-sorting the dependency list. If desired we can split it out into its own commit or remove it from the PR entirely.